src/otx/algorithms/visual_prompting/tasks/openvino.py

```diff
@@ -178,6 +194,63 @@ def await_all(self) -> None:
         self.model["decoder"].await_all()
 
 
+class OTXOpenVinoDataLoader:
+    """DataLoader implementation for VisualPromptingOpenVINOTask."""
+
+    def __init__(
+        self,
+        dataset: Any,
+        inferencer: OpenVINOVisualPromptingInferencer,
+        shuffle: bool = True,
+        is_encoder: bool = True,
+        output_model: Optional[ModelEntity] = None,
+    ):
+        self.dataset = dataset
+        self.inferencer = inferencer
+        self.shuffler = None
+        if shuffle:
+            self.shuffler = list(range(len(dataset)))
+            random.shuffle(self.shuffler)
+
+        self.is_encoder = is_encoder
+        self.target_length = self.inferencer.model["image_encoder"].orig_width
+        if not self.is_encoder:
+            core = ov.Core()
+            compressed_model = core.read_model(
+                output_model.get_data("visual_prompting_image_encoder.xml"),
+                output_model.get_data("visual_prompting_image_encoder.bin"),
+            )
+            self.compressed_model = core.compile_model(
+                model=compressed_model, device_name=inferencer.model["image_encoder"].inference_adapter.device
+            )
+
+    def __getitem__(self, index: int):
+        """Get item from dataset."""
+        if self.shuffler is not None:
+            index = self.shuffler[index]
+
+        items = self.dataset[index]
+        images, _, prompts = self.inferencer.pre_process(items)
+        processed_image = ResizeLongestSide.apply_image(images["images"][0], self.target_length).transpose(2, 0, 1)
```

Comment:
Why all this preprocess operations are implicitly added to dataset? They could be moved to pre_process, and executed only when a special flag was passed, like `self.inferencer.pre_process(items, extra_processing=True)`

Reply:
You mean, the below modification is better instead of using the function from `ResizeLongestSide`, right?

```python
# https://github.com/openvinotoolkit/training_extensions/blob/releases/1.4.0/src/otx/algorithms/visual_prompting/adapters/openvino/model_wrappers/openvino_models.py#L47-L51
def preprocess(self, inputs: np.ndarray, extra_processing: bool = True) -> Tuple[Dict[str, np.ndarray], Dict[str, Any]]:
    """Update meta for image encoder."""
    dict_inputs, meta = super().preprocess(inputs)
    if extra_processing:
        dict_inputs["images"] = self.apply_image(dict_inputs["images"][0], self.target_length).transpose(2, 0, 1)[None]
    meta["resize_type"] = self.resize_type
    return dict_inputs, meta

def apply_image(self, ...):
    ...
```

Reply:
I don't know about ResizeLongestSide, I said this calls sequence can be moved to the inferencer, since it's also a kind of preprocessing.

Reply:
Done.